File: Problem Solver/src/Problem005.cpp
#include "Problem005.h"
#include <iostream>
#include <string>

using namespace std;

Problem005::Problem005()
{
}

Problem005::~Problem005()
{
}


void Problem005::Solve()
{

    unsigned long long ans = 22;
    bool div = false;

    do
    {
notdiv:
        ans = ans + 2;
        for (long long i = 3; i <= 20; i++)
        {
            if (ans % i != 0)
            {
                goto notdiv;
            }
        }
        div = true;
    }
    while (div == false);



    cout << to_string(ans) << endl;
}

File: ProblemSolver/ProblemSolver/src/Problem005.cs
/*
Smallest multiple
Problem 5
2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.

What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ProblemSolver.src
{
    class Problem005 : SolvedProblems
    {
        public override void Solve()
        {
            ulong ans = 22;
            bool div = false;

            do
            {
            notdiv:
                ans += 2;

                for (ulong i = 3; i <= 20; i++)
                {
                    if (ans % i != 0)
                    {
                        goto notdiv;
                    }
                }
                div = true;
            } while (!div);

            EulerHelper.printAns(ans.ToString());
        }
    }
}
